packages/aws-cdk-lib/aws-codebuild/README.md

```diff
@@ -512,6 +512,59 @@ const fleet = new codebuild.Fleet(this, 'MyFleet', {
 });
 ```
 
+### Custom instance types
+You can use [specific EC2 instance
+types](https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-compute-types.html#environment-reserved-capacity.instance-types)
+for your fleet by setting the `computeType` to `CUSTOM_INSTANCE_TYPE`.  This
+allows you to specify the `instanceType` in `computeConfiguration`. Only certain
+EC2 instance types are supported; see the linked documentation for details.
+
+```ts
+import { Size } from 'aws-cdk-lib';
+
+const fleet = new codebuild.Fleet(this, 'MyFleet', {
+  baseCapacity: 1,
+  computeType: codebuild.FleetComputeType.CUSTOM_INSTANCE_TYPE,
+  environmentType: codebuild.EnvironmentType.LINUX_CONTAINER,
+  computeConfiguration: {
+    instanceType: ec2.InstanceType.of(ec2.InstanceClass.T3, ec2.InstanceSize.MEDIUM),
+    // By default, 64 GiB of disk space is included. Any value optionally
+    // specified here is _incremental_ on top of the included disk space.
+    disk: Size.gibibytes(10),
+  },
+});
+```
+
+### VPCs
+The same considerations that apply to [Project
+VPCs](#definition-of-vpc-configuration-in-codebuild-project) apply to Fleets.
+However, when a build runs on a Fleet reserved instance, any VPC configuration
+on the Project is not used - only the VPC configured on the Fleet has an effect.
```

Comment:
The docs are not very clear about this, but this is my understanding.  I am trying to verify this experimentally but my CodeBuild access got suspended 🥲, I guess because I was creating and destroying so many Fleets without really using them.  I am waiting for support to restore access...

I am similarly trying to verify what the Fleet Role is used for. It's not clear whether it's used during builds instead of the Role on the Project, or whether it's just used by CodeBuild to attach network interfaces... if it's only the latter, maybe there's no point in exposing it from Fleet?

Reply:
I decided to leave it, as I think it's fine _conceptually_ to expose this single Role, even if users can't currently do anything useful with it.